platform/linux-generic/odp_schedule_basic.c

```diff
@@ -1686,8 +1686,15 @@ static inline int schedule_loop_sleep(odp_queue_t *out_queue, uint64_t wait,
 			continue;
 		}
 
-		if (sleep)
-			nanosleep(&sched->powersave.sleep_time, NULL);
+		if (sleep && next) {
+			uint64_t sleep_time = _ODP_MIN(sched->powersave.sleep_time, next);
+			struct timespec ts = {
+				sleep_time / 1000000000,
+				sleep_time % 1000000000,
+			};
```

Comment:
I think sleep longer than 1sec does not need to be supported. tc_sec could be always 0, and tv_nsec value could be just masked with 0x1FFF FFFF to ensure that it's <= 0.536 sec.

Reply:
It's even easier to just cap the conf variable to less than one second. In v2.